Mathlib/RingTheory/MvPowerSeries/PiTopology.lean

```diff
@@ -0,0 +1,228 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos Fernández
+-/
+
+import Mathlib.RingTheory.Nilpotent.Defs
+import Mathlib.RingTheory.MvPowerSeries.Basic
+import Mathlib.Topology.Algebra.InfiniteSum.Constructions
+import Mathlib.Topology.Algebra.Ring.Basic
+import Mathlib.Topology.Algebra.UniformGroup.Basic
+import Mathlib.Topology.UniformSpace.Pi
+
+/-! # Product topology on multivariate power series
+
+Let `R` be with `Semiring R` and `TopologicalSpace R`
+In this file we define the topology on `MvPowerSeries σ R`
+that corresponds to the simple convergence on its coefficients.
+It is the coarsest topology for which all coefficient maps are continuous.
+
+When `R` has `UniformSpace R`, we define the corresponding uniform structure.
+
+This topology can be included by writing `open scoped MvPowerSeries.WithPiTopology`.
+
+When the type of coefficients has the discrete topology,
+it corresponds to the topology defined by [bourbaki1981], chapter 4, §4, n°2.
+
+It is *not* the adic topology in general.
+
+## Main results
+
+- `MvPowerSeries.WithPiTopology.tendsto_pow_zero_of_constantCoeff_nilpotent`,
+`MvPowerSeries.WithPiTopology.tendsto_pow_zero_of_constantCoeff_zero`: if the constant coefficient
+of `f` is nilpotent, or vanishes, then the powers of `f` converge to zero.
```

Comment:
As indicated by the syntax highlighting in vscode, you are missing the indentation here
```suggestion
  `MvPowerSeries.WithPiTopology.tendsto_pow_zero_of_constantCoeff_zero`: if the constant coefficient
  of `f` is nilpotent, or vanishes, then the powers of `f` converge to zero.
```
Same below